src/Service/Services/OpenAPI/OpenApiDocumentor.cs

```diff
@@ -0,0 +1,708 @@
+            };
+
+            return responsePayload;
+        }
+
+        /// <summary>
+        /// Builds the schema objects for all entities present in the runtime configuration.
+        /// Two schemas per entity are created:
+        /// 1) {EntityName}      -> Primary keys present in schema, used for request bodies (excluding GET) and all response bodies.
+        /// 2) {EntityName}_NoPK -> No primary keys present in schema, used for POST requests where PK is autogenerated.
+        /// Schema objects can be referenced elsewhere in the OpenAPI document with the intent to reduce document verbosity.
+        /// </summary>
+        /// <returns>Collection of schemas for entities defined in the runtime configuration.</returns>
+        private Dictionary<string, OpenApiSchema> CreateComponentSchemas()
+        {
+            Dictionary<string, OpenApiSchema> schemas = new();
+
+            foreach (string entityName in _metadataProvider.EntityToDatabaseObject.Keys.ToList())
+            {
+                // Entities which disable their REST endpoint must not be included in
+                // the OpenAPI description document.
+                if (_runtimeConfig.Entities.TryGetValue(entityName, out Entity? entity) && entity is not null)
+                {
+                    if (entity.GetRestEnabledOrPathSettings() is bool restEnabled)
+                    {
+                        if (!restEnabled)
+                        {
+                            continue;
+                        }
+                    }
+                }
+
+                SourceDefinition sourceDefinition = _metadataProvider.GetSourceDefinition(entityName);
+                List<string> columns = /*sourceDefinition is null ? new List<string>() : */sourceDefinition.Columns.Keys.ToList();
+
+                // create component for FULL entity with PK.
+                schemas.Add(entityName, CreateComponentSchema(entityName, fields: columns));
+
+                // create component for entity with no PK
+                // get list of columns - primary key columns then optimize
+                foreach (string primaryKeyColumn in sourceDefinition.PrimaryKey)
+                {
+                    columns.Remove(primaryKeyColumn);
+                }
+
+                // create component for TABLE (view?) NOT STOREDPROC entity with no PK.
+                DatabaseObject dbo = _metadataProvider.EntityToDatabaseObject[entityName];
+                if (dbo.SourceType is not SourceType.StoredProcedure or SourceType.View)
+                {
+                    schemas.Add($"{entityName}_NoPK", CreateComponentSchema(entityName, fields: columns));
+                }
+            }
+
+            return schemas;
+        }
+
+        /// <summary>
+        /// Creates the schema object for an entity.
+        /// </summary>
+        /// <param name="entityName">Name of the entity.</param>
+        /// <param name="fields">List of mapped (alias) field names.</param>
```

Comment:
Looking at the caller of this function - these fields seem to be list of original columns - NOT the mapped ones - since they are the ones coming from the sourceDefinition.

Could you please double check?

Reply:
correct. Updated the callers to use the helper GetExposedColumnNames() and past the list returned from that function to this one here.